src/datasources/errors/http-error-factory.ts

```diff
@@ -29,8 +27,12 @@ export class HttpErrorFactory {
   }
 }
 
-function isNetworkResponseError(
-  error: NetworkError | Error,
-): error is NetworkResponseError {
-  return 'status' in error && error.status >= 400 && error.status < 600;
+function isNetworkResponseError(error: unknown): error is NetworkResponseError {
+  return (
+    isObject(error) &&
+    'status' in error &&
+    typeof error.status === 'number' &&
+    error.status >= 400 &&
+    error.status < 600
+  );
```

Comment:
This can be reduced when we [restructure the errors thrown in the network client](https://github.com/safe-global/safe-client-gateway/blob/dd79a201e2aed3031c2a862f4390cf9ffa1f0fd5/src/datasources/network/network.module.ts#L23).